src/NzbDrone.Core/Download/Clients/Porla/PorlaProxy.cs

```diff
@@ -0,0 +1,234 @@
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using NLog;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Common.Http;
+using NzbDrone.Common.Serializer;
+using NzbDrone.Core.Download.Clients.Porla.Models;
+
+namespace NzbDrone.Core.Download.Clients.Porla
+{
+    #nullable enable
+    public interface IPorlaProxy
+    {
+        // sys
+
+        PorlaSysVersions GetSysVersion(PorlaSettings settings);     // sys.versions
+
+        // fs
+            // fs.space
+
+        // sessions
+
+        ReadOnlyCollection<PorlaSession> ListSessions(PorlaSettings settings);      // sessions.list
+        void PauseSessions(PorlaSettings settings);                                 // sessions.pause
+        void ResumeSessions(PorlaSettings settings);                                // sessions.resume
+        PorlaSessionSettings GetSessionSettings(PorlaSettings settings);           // sessions.settings.list
+
+        // presets
+
+        ReadOnlyDictionary<string, PorlaPreset> ListPresets(PorlaSettings settings);    // presets.list
+
+        // torrents
+
+        PorlaTorrent AddMagnetTorrent(PorlaSettings settings, string uri, IList<string>? tags = null);             // torrents.add
+        PorlaTorrent AddTorrentFile(PorlaSettings settings, byte[] fileContent, IList<string>? tags = null);       // torrents.add
+        void RemoveTorrent(PorlaSettings settings, bool removeData, PorlaTorrent[] pts);        // torrents.remove
+
+        // torrents.move
+
+        void PauseTorrent(PorlaSettings settings, PorlaTorrent pt);     // torrents.pause
+        void ResumeTorrent(PorlaSettings settings, PorlaTorrent pt);    // torrents.resume
+        ReadOnlyCollection<PorlaTorrentDetail> ListTorrents(PorlaSettings settings, int page = 0, int size = int.MaxValue);      // torrents.list
+
+        // torrents.recheck
+        // torrents.files.list
+        // torrents.metadata.list
+        // torrents.trackers.list
+
+        // torrents.peers
+            // torrents.peer.add
+            // torrents.peer.list
+
+        // torrents.properties
+            // torrents.properties.get
+            // torrents.properties.set
+    }
+
+    public class PorlaProxy : IPorlaProxy
+    {
+        private readonly IHttpClient _httpClient;
+        private readonly Logger _logger;
+
+        public PorlaProxy(IHttpClient httpClient, Logger logger)
+        {
+            _httpClient = httpClient;
+            _logger = logger;
+        }
+
+        private T ProcessRequest<T>(PorlaSettings settings, string method, params object?[] parameters)
+        {
+            var baseUrl = HttpRequestBuilder.BuildBaseUrl(settings.UseSsl, settings.Host, settings.Port, settings.UrlBase);
+
+            var jwt = settings.InfinteJWT ??= string.Empty;
+            var apiurl = settings.ApiUrl ??= string.Empty;
+
+            // this block will run a lot, don't want to be too noisy in the logs
```

Comment:
Do we need it at all? An empty if and else won't do anything.

Reply:
I guess the idea is for the future when they do implement JWT's we need to deal with it here.

What I actually want to do is just log what mode of JWT is currently being used.

Am i allowed to log messages in the ClientSettings.cs ?